### PR TITLE
 Add the KMWP & DKTC dataset.

### PR DESCRIPTION
**Pull request by sooftware**
Add the DKTC dataset.
- https://github.com/tunib-ai/DKTC

### CODE REVIEW

File: datasets/dktc/README.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+- 10K<n<100K
+source_datasets:
+- original
+task_categories:
+- text-classification
+task_ids:
+- multi-label-classification
+---
+
+# Dataset Card for [Needs More Information]
```
Comment (lhoestq) [inline datasets/dktc/README.md:22]:
```suggestion
# Dataset Card for Dataset of Korean Threatening Conversations
```

File: datasets/dktc/README.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+multilinguality:
+- monolingual
+size_categories:
+- 10K<n<100K
+source_datasets:
+- original
+task_categories:
+- text-classification
+task_ids:
+- multi-label-classification
```
Comment (lhoestq) [inline datasets/dktc/README.md:19]:
```suggestion
- multi-label-classification
pretty_name: Dataset of Korean Threatening Conversations
```

File: datasets/dktc/README.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+### Languages
+
+Korean
+
+## Dataset Structure
+
+### Data Instances
+
+{'class': 0,
+'text': '이 약이 뭔 거 같아? \n뭔데요.?\n뭐긴 뭐야 네 엄마 저승길 보내주는 약이지.\n네? 그게 무슨.\n네 엄마 지금 저 빌딩에 있지? 그리고 저 빌딩에선 시음회가 열리고 있고.\n그게.무슨.\n이미 인력은 다 배치해놨어. 저 음료를 마시면 네 엄마는.\n아 안 돼!!\n네 엄마 살리고 싶으면 내가 시키는 대로 해.\n멈춰요 알겠으니까 그만하라고!!\n내가 시키는 대로 해!!\n알겠다고요!!! 제발 그만'}
```
Comment (lhoestq) [inline datasets/dktc/README.md:72]:
````suggestion
Here is an example from the dataset:
```
{'class': 0,
'text': '이 약이 뭔 거 같아? \n뭔데요.?\n뭐긴 뭐야 네 엄마 저승길 보내주는 약이지.\n네? 그게 무슨.\n네 엄마 지금 저 빌딩에 있지? 그리고 저 빌딩에선 시음회가 열리고 있고.\n그게.무슨.\n이미 인력은 다 배치해놨어. 저 음료를 마시면 네 엄마는.\n아 안 돼!!\n네 엄마 살리고 싶으면 내가 시키는 대로 해.\n멈춰요 알겠으니까 그만하라고!!\n내가 시키는 대로 해!!\n알겠다고요!!! 제발 그만'}
```
````

File: datasets/dktc/README.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+### Data Fields
+
+{'class': Value(dtype='int32', id=None),
+ 'text': Value(dtype='string', id=None)}
```
Comment (lhoestq) [inline datasets/dktc/README.md:77]:
```suggestion
The dataset has two fields:
- **class**: the classification label of the text: threat (0), extortion (1), workspace (2) or etc (3)
- **text**: text content as a string
```

File: datasets/dktc/README.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+### Data Splits
+
+- Train
```
Comment (lhoestq) [inline datasets/dktc/README.md:81]:
```suggestion
- The train spit has 3950 samples, divided into several classes:
```

File: datasets/dktc/README.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+### Citation Information
+
+@misc{DKTC
+  author       = {Cho, Soyoung and Ha, Sangchun and Ryu, Myeonghyeon and
+                  Keum, Bitna and Park, Kyubyong},
+  title        = {DKTC, Dataset of Korean Threatening Conversations},
+  howpublished = {https://github.com/tunib-ai/DKTC},
+  year         = {2022},
+}
```
Comment (lhoestq) [inline datasets/dktc/README.md:149]:
````suggestion
```bibtex
@misc{DKTC
  author       = {Cho, Soyoung and Ha, Sangchun and Ryu, Myeonghyeon and
                  Keum, Bitna and Park, Kyubyong},
  title        = {DKTC, Dataset of Korean Threatening Conversations},
  howpublished = {https://github.com/tunib-ai/DKTC},
  year         = {2022},
}
```
````

File: datasets/dktc/dktc.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+class Dktc(datasets.GeneratorBasedBuilder):
+    """Dataset of Korean Threatening Conversations"""
+
+    VERSION = datasets.Version("1.0.0")
+
+    def _info(self):
+
+        features = datasets.Features(
+            {
+                "class": datasets.Value("int32"),
```
Comment (lhoestq) [inline datasets/dktc/dktc.py:57]: Here you can use the `ClassLabel` feature type to specify the mapping between the label ids (integers) and the label names (strings):
```suggestion
                "class": datasets.ClassLabel(names=["threat", "extortion", "workspace", "etc"]),
```

You'll need to regenerate the dataset_infos.json file after this change

**Review by lhoestq — COMMENTED** [review]
Hi ! Thanks for adding these datasets :)

I've reviewed the dataset card and the dataset script for DKTC - good job !
I just left some minor comments.

Can you please remove the KMWP files in this PR and open another PR to add KMWP, if you don't mind ? This is simpler for reviewing

**Comment by sooftware** [comment]
I reflect your review. cc. @lhoestq

**Comment by sooftware** [comment]
Ah sorry, I missed KMWP comment, wait.

**Comment by sooftware** [comment]
I request 2 new pull requests. #3569 #3570

**Pull request closed without merging**